### PR TITLE
[CCB] | MultiClient-CommitTimestampGetter refactor - part 1

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
Refactor CommitTimestampGetter to accommodate the flow for batching getCommitTimestamps requests across multiple clients.

**Implementation Description (bullets)**:

- Extract interface for CommitTimestampGetter - the only implementation right now is the BatchingCommitTimestampGetter which was the default request batching mode per client.
- Does not actually change anything in the current flow.

**Testing (What was existing testing like?  What have you done to improve it?)**:
Existing tests should suffice

**Concerns (what feedback would you like?)**:
Does this refactor break something?

**Where should we start reviewing?**:
CommitTimestampGetter

**Priority (whenever / two weeks / yesterday)**:
EOD today would be great

### CODE REVIEW

File: lock-api/src/main/java/com/palantir/lock/client/BatchingCommitTimestampGetter.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import com.palantir.lock.watch.TransactionUpdate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import org.immutables.value.Value;
+
+final class BatchingCommitTimestampGetter implements CommitTimestampGetter {
```
Comment (Jolyon-S) [inline lock-api/src/main/java/com/palantir/lock/client/BatchingCommitTimestampGetter.java:39]: [Sanity check]: This is the single namespace batching commit timestamp getter, right? In other words, this does _not_ batch between namespaces?

**Review by Jolyon-S — COMMENTED** [review]
Just a sanity check (because I can never keep up with naming conventions...) but I think this looks pretty much fine otherwise.

File: lock-api/src/main/java/com/palantir/lock/client/BatchingCommitTimestampGetter.java (hunk 1 above)
Comment (sudiksha27) [reply lock-api/src/main/java/com/palantir/lock/client/BatchingCommitTimestampGetter.java:43]: That is correct. Added docs on this class to say the same.

**Review by jeremyk-91 — APPROVED** [review]
🕐  0:02 🏁  -:-- Jolyon reviewed

👍  looks good!

**Pull request merged**